src/convertToObject.js

```diff
@@ -3,16 +3,28 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+
+  const temporality = sourceString.split(`;`).map((a) => a.split(':'));
+
+  for (const ch of temporality) {
+    if (ch[1] !== undefined) {
+      result[ch[0].trim()] = ch[1].trim();
+    }
+  }
```

Comment:
create a result object using reduce method instead of `for` loop

Reply:
how i can use  if (ch[1] !== undefined) in reduce?

Reply:
Please, take a look at this example and try to rewrite your code according to it 

```
return sourceString.split(';')
    .map(element => element.split(':')).filter(element => element.length === 2)
    .reduce((prev, current) => {
      prev[current[0].trim()] = current[1].trim();

      return prev;
    }, {});
```